### ISSUE TITLE
Idiotproof instructions for MarGo: Missing required environment variables: GOPATH

### ISSUE THREAD

**Issue by hickford** [DisposaBoy/GoSublime#310]
Hi. I installed this package. Now whenever I open Sublime Text (any file), this error message appears in the console

> MarGo: Missing required environment variables: GOPATH  
> See the `Quirks` section of USAGE.md for info

and a document `usage.MD` spontaneously opens.

How can I fix this? I tried to follow the instructions, but I'm new to both Go and Sublime Text , and don't quite understand what to do . I opened `Packages\User\GoSublime.sublime-settings` but it was blank. Also found `GoSublime\GoSublime.sublime-settings` but that only mentioned GOPATH offhand.

This is my fault for being an idiot, but would it be possible to idiotproof the instructions? Or better, make the package work out the box, if that's possible.

If it helps, I installed Go at `C:\Go`. Is that what I should put?

**Comment by GreenRaccoon23** [comment]
I'm still having trouble with this. (I'm using Arch Linux). When I place into `~/.config/sublime-text-3/Packages/User/GoSublime.sublime-settings` this setting:
```
{
	"env": { "GOPATH": "$HOME/go:$GS_GOPATH" }
}
```
I receive the error `MarGo: Missing required environment variables: GOROOT`. So then if I change that to file to this:
```
{
	"env": { "GOPATH": "$HOME/go:$GS_GOPATH" },
	"env": { "GOROOT": "/usr/lib/go:$GS_GOPATH" }
}
```
I get the error `MarGo: Missing required environment variables: GOPATH`.

I've exported both variables in my path as well (in my `.bashrc` file). I'm sure I'm making a little syntax mistake, right?

**Comment by chmac** [comment]
Correct, you are making a syntax mistake, the correct syntax is:

    {
        "env": {
            "GOPATH": "$HOME/go:$GS_GOPATH",
            "GOROOT": "/usr/lib/go:$GS_GOROOT"
        }
    }

In your example, you created two values both with the key `env` which means the second overwrote the first.

**Comment by soroushjp** [comment]
For anyone else with zsh on Mac OS X:

My GOPATH and GOROOT are set but for some reason MarGo wasn't seeing them after I installed zsh. Putting them in my GoSublimge -> Settings – User (not general Sublime User settings) fixed this issue.

**Comment by denji** [comment]
How can I ignore this error? I don't want to run right now to install go to another machine, Sublime begins to slow down and recursively bombarding me with messages:
```
MarGo: Missing required environment variables: GOROOT
See the `Quirks` section of USAGE.md for info
```

**Comment by DisposaBoy** [comment]
you can uninstall GoSublime or add it your list of ignored packages in Sublime Text

**Comment by liujb** [comment]
@raizul8 thx.

**Comment by GreenRaccoon23** [comment]
@chmac Whoa I never thanked you! Here's a belated thanks!!!

**Comment by abhijeeg** [comment]
I need to setup the GOPATH on Windows 10 machine.
Here is the path setup in my environment variable: C:\Go\bin
How should I updated it in SublimeTXT2

**Comment by luck02** [comment]
My overall experience of getting this plugin working is that the instructions are inadequate.   A link to the go environment setup: https://golang.org/doc/code.html would be good... With the standard disclaimer: "If you don't know what GOPATH is go read this".  

For me the real issue was that the usage.md file stated: 

```
See the documentation for the env and/or shell setting, both documented in the default settings file Packages/User/GoSublime.sublime-settings
```

When in fact there's no documentation on those settings in any of the files.  I had to google it up and it wasn't an instant hit.  I did find this article:  http://www.wolfe.id.au/2015/03/05/using-sublime-text-for-go-development/ which explained the process.
